zilliqa/src/exec.rs

```diff
@@ -673,23 +676,101 @@ impl State {
         value: u128,
     ) -> Result<u64> {
         let gas_price = gas_price.unwrap_or(GAS_PRICE);
-        let gas = gas.unwrap_or(BLOCK_GAS_LIMIT);
 
+        let mut right = gas.unwrap_or(MAX_EVM_GAS_LIMIT).0;
+        let mut left = lower_bound_gas_estimate(to_addr, &data);
```

Comment:
nit: I'd find it clearer if these were called `min` and `max`.

Reply:
Can we also add in a comment if these are inclusive or exclusive? (I think `left` is inclusive and `right` is exclusive but I'm not sure?)